url85940d3d8e_ccp999_Expr4j_tgz-pJ8-org_boris_expr_util_ExprEvaluatorJ8/src/org/boris/expr/parser/ExprLexer.java

```diff
@@ -0,0 +1,266 @@
+/*******************************************************************************
+ * This program and the accompanying materials
+ * are made available under the terms of the Common Public License v1.0
+ * which accompanies this distribution, and is available at 
+ * http://www.eclipse.org/legal/cpl-v10.html
+ * 
+ * Contributors:
+ *     Peter Smith
+ *******************************************************************************/
+package org.boris.expr.parser;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.Reader;
+import java.io.StringReader;
+
+public class ExprLexer
+{
+    private TokenReader reader;
+    private int lastChar;
+
+    public ExprLexer(BufferedReader reader) {
+        this.reader = new TokenReader(reader);
+    }
+
+    public ExprLexer(Reader reader) {
+        this(new BufferedReader(reader));
+    }
+
+    public ExprLexer(String str) {
+        this(new StringReader(str));
+    }
+
+    public ExprToken next() throws IOException {
+        if (lastChar == 0 || Character.isWhitespace(lastChar))
+            lastChar = reader.ignoreWhitespace();
+
+        return readToken();
+    }
+
+    private ExprToken readToken() throws IOException {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method contains 54 lines of code, not including blank lines or lines with only comments, Java punctuation characters, identifiers, or literals. By comparison, 98% of the methods in the CodeGuru reference dataset contain fewer lines of code. Large methods might be difficult to read and have logic that is hard to understand and test. We recommend that you simplify this method or break it into multiple methods.